### PR TITLE
backend: uses session cookies for authentication

### PR DESCRIPTION
**Pull request by forschoolandnothingelse**
The previous implementation of the backend saved passwords in plaintext
on the clientside. This was really bad for a number of reasons, not
least of which is it just didn't work on a ton of browsers.

The new backend uses an /auth route to first authenticate a user. It
expects a JSON object of the form:
{
    email: user@email.com,
    pwd: qwerty
}
From here it either authenticates the user's session and returns a JSON
object with the "error" field set to null, or does not authenticate the
user's session and returns an object with the "error" field set to an
error string.

This also changes the way the /{create,read,update,delete}User routes
works. Now all user routes are just under /user. The way we do CRUD for
users is using HTTP methods, POST for create, GET for read, PUT for
update, and DELETE for delete. This should enforce not making every API
call a POST for no good reason.

This also enforces better security since a user can only modify their
own user entry as determined for the session rather than the user
themselves.

The way user encryption works changes as well. Previously because the
user always sent their password, the password was used as the key. Now a
special encryption key is saved. The key is a simple encrypt-then-MAC of
the user's password and the sessionSecret. This will break all
previously stored data in Bujo, but makes for better security in the
long run.

Of course, we need to add an envvar for the sessionSecret under the key
SESSION_SECRET, or deployment to production will fail.

---

Updates the frontend to use the new backend API in the previous commit.
This required first getting auth cookies before getting users. This
commit also removes saving passwords on the clientside.

### CODE REVIEW

**Review by waynarbocangel — APPROVED** [review]
Everything seems reasonable and correctly refactored. It looks like everything should be working correctly and the routes are properly redefined.

File: docs/global.html
Comment (ehsonp) [inline docs/global.html:182]: WHy is this file format so spaced?

**Review by ehsonp — APPROVED** [review]
Im confused why the one file with 16,000 deletions has so many empty lines but I approve

**Comment by forschoolandnothingelse** [comment]
> Im confused why the one file with 16,000 deletions has so many empty lines but I approve

I'm pretty sure you're supposed to run `npm run jsdoc` before making a commit, so that was what I was doing.

---

I'm going to hold off on merging this since I don't want to break the build because the backend server is missing env vars. @waynarbocangel can merge this when he thinks the server is ready.

**Pull request merged**
